src/reanimated2/layoutReanimation/sharedTransitions/ProgressTransitionManager.ts

```diff
@@ -72,8 +72,19 @@ export class ProgressTransitionManager {
           global.ProgressTransitionRegister.onTransitionEnd();
         }
       );
-      if (Platform.OS === 'ios') {
-        // required by modals
+
+      if (Platform.OS === 'android') {
+        // onFinishTransitioning event is available only on Android and
+        // is used to handle closing modals
+        eventHandler.onDisappear = registerEventHandler(
+          'onFinishTransitioning',
+          () => {
+            'worklet';
+            global.ProgressTransitionRegister.onAndroidFinishTransitioning();
+          }
```

Comment:
Is it possible to pass this function directly here?
```suggestion
          global.ProgressTransitionRegister.onAndroidFinishTransitioning,
```

Reply:
I will check it!

Reply:
Unfortunately, it doesn't work because this lambda is created on the JS thread, where `global.ProgressTransitionRegister` is unavailable :/